src/OSPSuite.SimModel/src/SimulationTask.cpp

```diff
@@ -222,6 +222,54 @@ void SimulationTask::CheckForNegativeValues(Species ** odeVariables, int numberO
 
 }
 
+void SimulationTask::MarkUsedParameters(Simulation * sim)
```

Comment:
Main routine. MArks all parameters as used, which are used in:
* ODE Variables (RHS or initial formula)
* Observers
* Event conditions
* Event assignment formulas